packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -223,13 +253,30 @@ async function updateRepoOptions(repo: Repo, context: Context) {
   console.log(`enable rebase? ${config.enableRebaseMerge}`);
   console.log(`enable squash? ${config.enableSquashMerge}`);
 
-  await context.github.repos.update({
-    name,
-    repo: name,
-    owner,
-    allow_merge_commit: false,
-    allow_rebase_merge: config.enableRebaseMerge,
-    allow_squash_merge: config.enableSquashMerge,
-  });
+  try {
+    await context.github.repos.update({
+      name,
+      repo: name,
+      owner,
+      allow_merge_commit: false,
+      allow_rebase_merge: config.enableRebaseMerge,
+      allow_squash_merge: config.enableSquashMerge,
+    });
+  } catch (err) {
+    const knownErrors = [
+      401, // bot does not have permission to access this repository.
+      403, // thrown if repo is archived.
```

Comment:
This is a fun one.  In the general sense - should bots ever run against archived repositories?  I'm wondering if we should do this check up front and then quietly exit if it is.